content/blog/oss-iac.md

```diff
@@ -0,0 +1,184 @@
+
+It helps with smaller projects too. Need to renovate that bathroom? That’s a lot simpler when everything already follows a specification. Who knows, there might be upgraded components available that meet your needs.
+
+While LOC is not the silver bullet when assessing software, it can be used as one component of approximating codebase complexity. As an example for this reason to use OSS, let’s look at the number of LOC in some very commonly used modules:
+
+1. [https://github.com/cloudposse/terraform-aws-vpc](https://github.com/cloudposse/terraform-aws-vpc) — 666 LOC
+2. [https://github.com/cloudposse/terraform-aws-alb—](https://github.com/cloudposse/terraform-aws-alb—) 1064 LOC
+3. [https://github.com/cloudposse/terraform-aws-eks-cluster](https://github.com/cloudposse/terraform-aws-eks-cluster) — 1224 LOC
+
+What’s more efficient? For you to write (and maintain!) those 2K lines of code for your organization or for you to consume those 3 modules and benefit from the hard work that others have put into making them great?
+
+## Objections to open-source IaC
+
+Of course, no solution is perfect. When I often suggest using OSS modules, I tend to hear these objections and will offer a brief rebuttal to each:
+
+#### “There are too many inputs to so-and-so module – [it feels like a leaky abstraction](https://en.wikipedia.org/wiki/Leaky_abstraction)”
+
+This can happen and there are good examples of OSS modules who are prime offenders. I wouldn’t recommend using them. Evaluate each module independently, review which inputs are required, and make an informed decision. OSS modules aren’t magic pixie dust–you still have to do your due diligence.
+
+#### “Using external modules sounds like dependency hell"
+
+Software is all about managing dependencies, whether OSS or not. The IaC ecosystem is not the Javascript ecosystem; If you use good modules, pin them to a specific version, and update when you need to then you won’t run into the dependency issues that you may have elsewhere.
+
+#### “We can’t use open-source in my organization because of XYZ reason”
+
+This is a sad reality for some folks and to those folks I suggest: Copy / pasta the open-source module into your codebase, abiding by the license. Plan to update it from time to time.
+
+#### “But I can generate all this IaC with AI… why wouldn’t I do that?”
+
+Did you take the time to understand the AI generated IaC code at a deep level? Did the person who reviewed it? If not, now you have a bunch of code that you need to maintain **and** it won’t be getting updates from the community. Congrats, you’ve increased your future maintenance burden!
+
+## How to evaluate good OSS IaC
+
+Above I suggest that you need to do your own due diligence to ensure that the modules that you use in your organization are quality. Here are some questions that you can ask to check the quality of any given OSS module:
+
+#### Check the feature support
```

Comment:
Maybe make it a numbered list? This way it's easier to follow the thoughts (IMO).

Reply:
I originally had it this way and then moved away from it. I think we keep it this way because we're expanding on each point and using headers for each here will allow someone to skim if they don't really care about this section, but want to see how we think about it.